packages/strapi-admin/admin/src/containers/ProfilePage/index.js

```diff
@@ -64,6 +75,83 @@ const ProfilePage = () => {
             })}
           </FormBloc>
         </ContainerFluid>
+
+        <ContainerFluid padding="18px 30px 18px 30px">
+          <Bloc>
+            <Padded top left right size="md">
+              <Padded top bottom size="sm">
+                <Padded top bottom size="sm">
+                  <Title>
+                    {formatMessage({ id: 'Settings.profile.form.section.password.title' })}
+                  </Title>
+                </Padded>
+              </Padded>
+
+              <Flex>
+                <div style={{ flex: 1 }}>
+                  <IntlInput
+                    label="Auth.form.password.label"
+                    type="password"
+                    autoComplete="new-password"
+                    validations={{}}
+                    error={formErrors.password}
+                    name="password"
+                    onChange={handleChange}
+                    value={get(modifiedData, 'password', '')}
+                  />
+                </div>
+
+                <Spacer />
+
+                <div style={{ flex: 1 }}>
+                  <IntlInput
+                    label="Auth.form.confirmPassword.label"
+                    type="password"
+                    validations={{}}
+                    error={formErrors.confirmPassword}
+                    name="confirmPassword"
+                    onChange={handleChange}
+                    value={get(modifiedData, 'confirmPassword', '')}
+                  />
+                </div>
+              </Flex>
+            </Padded>
+          </Bloc>
+        </ContainerFluid>
+
+        <ContainerFluid padding="18px 30px 18px 30px">
+          <Bloc>
+            <Padded top left right bottom size="md">
+              <Padded top bottom size="sm">
+                <Padded top bottom size="sm">
+                  <Title>
+                    {formatMessage({ id: 'Settings.profile.form.section.experience.title' })}
+                  </Title>
+                </Padded>
+              </Padded>
+
+              <ProfilePageLabel htmlFor="">Interface language</ProfilePageLabel>
+
+              <div style={{ width: 'calc(50% - 3rem)' }}>
+                <Select
+                  aria-labelledby="interface-language"
+                  selectedValue={currentLanguage}
+                  onChange={selectLanguage}
```

Comment:
If I understood correctly, the app language will be stored in DB, therefore, we need to change the language only after the form has been submitted successfully.

Reply:
I think that's the intended behaviour 😊

Reply:
It seems weird as the setting is stored in DB, IMO the switch needs to be done after the request has been sent, otherwise it changes your app language and when you refresh your app, the language will still be the one from your db

Reply:
Yes sure, but the call is not made for now 😊 basically it could eventually look like:

```jsx
<Select
  onChange={nextLocale => 
    updateRemoteLocale(nextLocale)
      .then(selectLanguage)
      .catch(() => /* do nothing*/)
/>
```